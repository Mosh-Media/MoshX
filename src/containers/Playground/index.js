import React, { useState } from "react";

import CodeEditor from "components/CodeEditor";
import MoHelmet from "components/shared/MoHelmet";

const Playground = () => {
  const initialState = {
    answer: "<button> I am a Button </button>",
    language: "jsx",
    element: "button",
    id: "0",
    isCorrect: false,
    question: `() => {
      return (
        <section>
          <h1>I love React ðŸ’™</h1>
        </section>
      );
    };`,
    isPlayground: true,
  };

  const [question, setQuestion] = useState(initialState);

  const handleOnChange = (userAnswer) => {
    if (userAnswer === "{}" || userAnswer === "") {
      return;
    }
    setQuestion({ ...question, question: userAnswer });
  };

  return (
    <>
      <MoHelmet
        title="Moskool - React frontend development playground"
        description="MoSkool - Try out free React development playground, master JavaScript, Html and Css without the fees"
      />
      <CodeEditor
        handleOnChange={(userAnswer) => handleOnChange(userAnswer)}
        md={12}
        sm={12}
        question={question}
      />
    </>
  );
};

export default Playground;
