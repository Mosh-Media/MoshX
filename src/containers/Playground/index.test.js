import React from "react";

import Playground from "./index.js";
import ShallowRenderer from "react-test-renderer/shallow";

const renderer = new ShallowRenderer();

describe("Playground Component", () => {
  test("should match the snapshot", () => {
    renderer.render(<Playground />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
