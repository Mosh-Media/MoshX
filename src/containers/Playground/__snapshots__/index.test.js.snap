// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playground Component should match the snapshot 1`] = `
<React.Fragment>
  <CodeEditor
    handleOnChange={[Function]}
    md={12}
    question={
      Object {
        "answer": "<button> I am a Button </button>",
        "element": "button",
        "id": "0",
        "isCorrect": false,
        "isPlayground": true,
        "language": "jsx",
        "question": "() => {
      return (
        <section>
          <h1>I love React ðŸ’™</h1>
        </section>
      );
    };",
      }
    }
    sm={12}
  />
</React.Fragment>
`;
